### ISSUE TITLE
LICENSE and CKAN reuse

### ISSUE THREAD

**Issue by chrismetcalf** [timwis/jkan#54]
Very cool work @timwis!

Important question to check on, however. It looks like you've licensed JKAN under the [MIT License](https://github.com/timwis/jkan/blob/gh-pages/LICENSE) (one of my favorites), but CKAN itself is [Affero GPL licensed](https://github.com/ckan/ckan#copying-and-license).

Did you reuse any CKAN code in JKAN? I'm interested in contributing to JKAN or creating a Socrata-backed version, but I want to make sure we're treading on safe ground with the license.

**Comment by waldoj** [comment]
CKAN is written in Python, while JKAN is all Markdown and JavaScript. :)

**Comment by chrismetcalf** [comment]
Sure, but there’s also CSS and page templates, which would still be covered
by the AGPL if they were lifted from CKAN :)

**Comment by waldoj** [comment]
Well, [here's all of the custom JKAN CSS](https://demo.jkan.io/css/main.css). :)

**Comment by chrismetcalf** [comment]
Hmm, taking a little tour through `_includes`, it looks like its just Bootstrap plus that, right? I trust you guys, and I'm not trying to be a stick in the mud, AGPL just scares me. :)

**Comment by waldoj** [comment]
Oh, only @timwis can _definitively_ answer this (I haven't contributed a single byte to this), but I don't see any sign that JKAN is based on CKAN code any more than DKAN is.

**Comment by chrismetcalf** [comment]
:+1: I'll wait for @timwis to chime in (he's probably busy being a CDO or something).

**Comment by timwis** [comment]
haha not too busy :P

First of all, a great point you've raised. When I started, I _almost_ began by cloning their markup/templates. I decided not to for sake of MVP/saving time, and I'm glad I did as I hadn't considered the implications of their license.

I can't think of a single ounce of code I borrowed from CKAN. The layout, terminology, and functionality are certainly inspired by CKAN (let alone the name), but none of the code, CSS, or markup come from it. I did use the schema that CKAN uses, but I actually just got off a call with the data.gov folks and we're going to switch JKAN to use the data.json schema tomorrow.

So it should all be okay, and I would definitely want to license the software in a way that would encourage what you described. I'm looking forward to seeing what you build!

I'll close this issue for now but feel free to reply or re-open.
